artiq/coredevice/ad9910.py

```diff
@@ -222,6 +223,20 @@ def set_phase_mode(self, phase_mode):
         """
         self.phase_mode = phase_mode
 
+    @kernel
+    def write16(self, addr, data):
+        """Write to 16 bit register.
+
+        :param addr: Register address
+        :param data: Data to be written
+        """
+        self.bus.set_config_mu(urukul.SPI_CONFIG, 8,
+                               urukul.SPIT_DDS_WR, self.chip_select)
+        self.bus.write(addr << 24)
+        self.bus.set_config_mu(urukul.SPI_CONFIG | spi.SPI_END, 16,
+                               urukul.SPIT_DDS_WR, self.chip_select)
+        self.bus.write(data)
```

Comment:
1. Did you actually test this? The `data` write should be MSB aligned.

2. This should be done in one xfer. Could you test the following?

```suggestion
        self.bus.set_config_mu(urukul.SPI_CONFIG | spi.SPI_END, 24,
                               urukul.SPIT_DDS_WR, self.chip_select)
        self.bus.write((addr << 24) | (data << 8))
```